### PR TITLE
feat: Add classNames and styles prop for card

### LINKED ISSUE #46260: Add className for inner child div's of components (i.e Card)
### What problem does this feature solve?

For a better usability with TailwindCSS, I suggest that components that has one inner child that wraps everything that it'll be easy to add className, for example the body of a Card.
Moreover I think antd can improve in manner of customizations, lot's of properties are still hidden even in v5.
If tailwindcss can be used with it, it'l be great.
sometime there is a need to use important over antd classes which I didn't still realize why it rise above the immediate className being applied.

Thank you!

### What does the proposed API look like?

`<Card className="items-strech"/>` 
Or if you don't to use the className prop maybe create something similar to the `bodyStyle`

### PR DESCRIPTION
**Pull request by zh-lx**
close #46260 

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

#46260 

### 💡 Background and solution

Add classNames and styles prop for card

### 📝 Changelog

Add classNames and styles prop for card

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Add classNames and styles prop for card      |
| 🇨🇳 Chinese |    Card 组件增加 classNames 和 styles 属性       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/card/Card.tsx [hunk 1]
```diff
@@ -45,12 +45,39 @@ export interface CardProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 't
...
+  };
+  styles?: {
+    head?: React.CSSProperties;
+    body?: React.CSSProperties;
+    extra?: React.CSSProperties;
+    title?: React.CSSProperties;
+    tabs?: React.CSSProperties;
+    actions?: React.CSSProperties;
+    cover?: React.CSSProperties;
+    headWrapper?: React.CSSProperties;
```
Comment (afc163) [inline components/card/Card.tsx:66]: headWrapper 是啥，这个和具体结构有关系，如果 html 重构有可能就不存在了，不建议放在语义化配置里。

**Comment by li-jia-nan** [comment]
feature branch

File: components/card/Card.tsx [hunk 2]
```diff
@@ -45,12 +45,37 @@ export interface CardProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 't
   activeTabKey?: string;
   defaultActiveTabKey?: string;
   tabProps?: TabsProps;
+  classNames?: {
+    head?: string;
+    body?: string;
+    extra?: string;
+    title?: string;
+    tabs?: string;
```
Comment (MadCcc) [inline components/card/Card.tsx:53]: 这个可以不加

File: components/card/Card.tsx [hunk 3]
```diff
@@ -45,12 +45,37 @@ export interface CardProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 't
   activeTabKey?: string;
   defaultActiveTabKey?: string;
   tabProps?: TabsProps;
+  classNames?: {
+    head?: string;
```
[the remark this answers is not in the document]
Comment (zh-lx) [reply components/card/Card.tsx]: 已修改

**Comment by li-jia-nan** [comment]
feature 分支

**Comment by afc163** [comment]
@MadCcc

File: components/card/Card.tsx [hunk 4]
```diff
@@ -139,25 +172,57 @@ const Card = React.forwardRef<HTMLDivElement, CardProps>((props, ref) => {
...
+  const coverClasses = classNames(`${prefixCls}-cover`, moduleClass('cover'));
+  const coverDom = cover ? (
+    <div className={coverClasses} style={moduleStyle('cover')}>
+      {cover}
+    </div>
+  ) : null;
+  const bodyClasses = classNames(`${prefixCls}-body`, moduleClass('body'));
+  const mergedBodyStyle: React.CSSProperties = {
+    ...moduleStyle('body'),
+    ...bodyStyle,
```
Comment (MadCcc) [inline components/card/Card.tsx:209]: 有 styles 之后 bodyStyle 就可以废弃了，加一个 warning，这里优先级也应该让 styles.body 覆盖 bodyStyle

Comment (MadCcc) [reply components/card/Card.tsx:209]: bodyStyle 就从文档 API 中去除

File: components/card/index.zh-CN.md [hunk 5]
```diff
@@ -56,6 +56,8 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*a-8zR6rrupgAAA
 | tabProps | [Tabs](/components/tabs-cn#tabs) | - | - |  |
 | title | 卡片标题 | ReactNode | - |  |
 | type | 卡片类型，可设置为 `inner` 或 不设置 | string | - |  |
+| classNames | 配置卡片内置模块的 className | `header?: string; body?: string; extra?: string; title?: string; actions?: string; cover?: string;` | - | 5.13.0 |
```
Comment (MadCcc) [inline components/card/index.zh-CN.md:59]: https://ant.design/components/button-cn#styles-%E5%92%8C-classnames-%E5%B1%9E%E6%80%A7
参考 Button 页，把语义化的属性单独列一个 table

Comment (zh-lx) [reply components/card/index.zh-CN.md]: done

File: components/card/Card.tsx (hunk 4 above)
Comment (zh-lx) [reply components/card/Card.tsx:220]: done

**Comment by li-jia-nan** [comment]
冲突看一下

**Comment by zh-lx** [comment]
> 冲突看一下

done!

**Comment by zh-lx** [comment]
@afc163 @MadCcc 

> Hi @zh-lx. Thanks for your contribution. The path `.github/` or `scripts/` and `CHANGELOG` is only maintained by team members. This current PR will be closed and team members will help on this.

I change the table count of card in `scripts/check-site.snap`. It directly be closed by bot. 😂

**Comment by zh-lx** [comment]
add a new PR #46811

**Pull request closed without merging**